### PR TITLE
[Method References] LPS-162265 Convert method references to field references in class extends BaseJSPProductNavigationControlMenuEntry

### PR DESCRIPTION
**Pull request by W2jx**
Hi @tinatian @dantewang , I convert method references to field references in class extends BaseJSPProductNavigationControlMenuEntry.

### CODE REVIEW

**Comment by W2jx** [comment]
Hi @catarineaguiar , Could you please help me check the test results when they finish ? Thanks!

File: modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseJSPProductNavigationControlMenuEntry.java [hunk 1]
```diff
@@ -72,9 +72,7 @@ public boolean includeIcon(
 			httpServletRequest, httpServletResponse, getIconJspPath());
 	}
 
-	public void setServletContext(ServletContext servletContext) {
-		_servletContext = servletContext;
-	}
+	protected abstract ServletContext getServletContext();
```
Comment (dantewang) [inline modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseJSPProductNavigationControlMenuEntry.java:75]: @W2jx If the `ServletContext` is only used in the base class to obtain the `RequestDispatcher`, I think you can just let the subclass implement an abstract method `getRequestDispatcher(String jspPath)`. So subclasses still reference the `ServletContext` but directly give the `RequestDispatcher` to parent.

Comment (dantewang) [reply modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseJSPProductNavigationControlMenuEntry.java:75]: cc @tinatian

**Review by dantewang — CHANGES_REQUESTED** [review]

File: modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseJSPProductNavigationControlMenuEntry.java (hunk 1 above)
Comment (W2jx) [reply modules/apps/product-navigation/product-navigation-control-menu-api/src/main/java/com/liferay/product/navigation/control/menu/BaseJSPProductNavigationControlMenuEntry.java:75]: I will fix it.

**Comment by catarineaguiar** [comment]
@W2jx no regressions found.

**Pull request closed without merging**